### ISSUE TITLE
Encoding Issue (VIP)

### ISSUE THREAD

**Issue by Diego-OT** [otland/forgottenserver#4560]
- [*] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [*] This issue is reproducible without changes to the C++ code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Add a player in VIP
2. Edit VIP
3. Write something like: filme de ação

### Expected behaviour
normal save

### Actual behaviour
[Error - executeQuery]

The code below seems to fix the error with saving correct in db, but not display correct after relogin (need to reverse the conversion):

// cip client use Latin-1 encoding (ISO-8859-1)
std::string iso_8859_1_to_utf8(std::string &str)
{
    std::string strOut;
    for (std::string::iterator it = str.begin(); it != str.end(); ++it)
    {
        uint8_t ch = *it;
        if (ch < 0x80) {
            strOut.push_back(ch);
        } else {
            strOut.push_back(0xc0 | ch >> 6);
            strOut.push_back(0x80 | (ch & 0x3f));
        }
    }
    return strOut;
}

I hope there is some better solution, thanks.

**Comment by amatria** [comment]
I spent some time evaluating this issue, and I don't think it should be marked as a "good first issue." I believe the underlying problem is profound and requires careful evaluation.

The issue at hand is that since #2403 (which addressed issue #2175), all text in the database has been stored using UTF-8 encoding. However, the data that comes from the client is Latin-1 encoded. This isn't problematic as long as the characters from the client stay within the ASCII range, as both Latin-1 and UTF-8 encode these characters the same way. However, if we receive from the client a string of text containing characters beyond the ASCII range (e.g., "ñññççç"), this is where data loss occurs, resulting in the situation described by the original poster. We end up storing in the database a set of characters that are not UTF-8 encoded.

One possible solution is to use the `CONVERT()` function in SQL to convert text from Latin-1 to UTF-8 before storing it in the database:

```sql
INSERT INTO table (utf8_column) VALUES (CONVERT('latin1_text_ñññççç' USING utf8));
```

Similarly, we can use `CONVERT()` to convert the UTF-8 encoded text back to Latin-1 encoding:

```sql
SELECT CONVERT(utf8_column USING latin1) AS latin1_column FROM table;
```

However, I'm hesitant to consider this solution as it may introduce compatibility issues with existing code, which currently assumes that strings are encoded in UTF-8 format. Another approach could involve addressing the encoding mismatch at the game protocol level, implementing explicit conversion mechanisms to handle text encoding conversions seamlessly during data transmission between the client and server.

Thoughts?

**Comment by amatria** [comment]
This triggers the issue too:

https://github.com/otland/forgottenserver/assets/34030065/51865a5b-75bf-42d0-bf53-f78235ff2f46
